src/OpenTelemetry.Collector.AspNet/Implementation/HttpInListener.cs

```diff
@@ -0,0 +1,144 @@
+            : base(name, tracer)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnStartActivity";
+
+            var context = HttpContext.Current;
+            if (context == null)
+            {
+                CollectorEventSource.Log.NullPayload(nameof(HttpInListener) + EventNameSuffix);
+                return;
+            }
+
+            if (this.options.RequestFilter != null && !this.options.RequestFilter(context))
+            {
+                CollectorEventSource.Log.RequestIsFilteredOut(activity.OperationName);
+                return;
+            }
+
+            var request = context.Request;
+            var requestValues = request.Unvalidated;
+
+            // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md
+            var path = requestValues.Path;
+
+            TelemetrySpan span;
+            if (this.options.TextFormat is TraceContextFormat)
+            {
+                this.Tracer.StartActiveSpanFromActivity(path, Activity.Current, SpanKind.Server, out span);
+            }
+            else
+            {
+                var ctx = this.options.TextFormat.Extract<HttpRequest>(
+                    request,
+                    (r, name) => requestValues.Headers.GetValues(name));
+
+                this.Tracer.StartActiveSpan(path, ctx, SpanKind.Server, out span);
+            }
+
+            if (span.IsRecording)
+            {
+                span.PutHttpHostAttribute(request.Url.Host, request.Url.Port);
+                span.PutHttpMethodAttribute(request.HttpMethod);
+                span.PutHttpPathAttribute(path);
+
+                span.PutHttpUserAgentAttribute(request.UserAgent);
+                span.PutHttpRawUrlAttribute(request.Url.ToString());
+            }
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnStopActivity";
+            var span = this.Tracer.CurrentSpan;
+
+            if (span == null || !span.Context.IsValid)
+            {
+                CollectorEventSource.Log.NullOrBlankSpan(nameof(HttpInListener) + EventNameSuffix);
+                return;
+            }
+
+            if (span.IsRecording)
+            {
+                var context = HttpContext.Current;
+                if (context == null)
+                {
+                    CollectorEventSource.Log.NullPayload(nameof(HttpInListener) + EventNameSuffix);
+                    return;
+                }
+
+                var response = context.Response;
+
+                span.PutHttpStatusCode(response.StatusCode, response.StatusDescription);
+
+                var routeData = context.Request.RequestContext.RouteData;
```

Comment:
null checks needed?

Reply:
Surprisingly, no: https://referencesource.microsoft.com/#system.web/HttpRequest.cs,238

I had them in there originally and removed after testing.